portal-backend/depmap/data_explorer_2/views.py

```diff
@@ -464,6 +470,64 @@ def get_labels_matching_context():
     return make_gzipped_json_response(labels_matching_context)
 
 
+@blueprint.route("/v2/context", methods=["POST"])
+@csrf_protect.exempt
+def evaluate_v2_context():
+    """
+    Get the full list of labels (in any dataset) which match the given context.
+    """
+    inputs = request.get_json()
+    context = inputs["context"]
+    dimension_type = context.get("dimension_type")
+    if dimension_type is None:
+        abort(400, "v2 Contexts must have a 'dimension_type' field")
+
+    # Load all dimension labels and ids
+    all_labels_by_id = get_all_dimension_labels_by_id(dimension_type)
+
+    # Evaluate each against the context
+    context_evaluator = ContextEvaluator(context, data_access.get_slice_data)
+    ids_matching_context = []
+    labels_matching_context = []
+    for given_id, label in all_labels_by_id.items():
+        if context_evaluator.is_match(given_id):
+            ids_matching_context.append(given_id)
+            labels_matching_context.append(label)
+
+    return make_gzipped_json_response(
+        {"ids": ids_matching_context, "labels": labels_matching_context,}
```

Comment:
This now returns _both_ labels and IDs for anything that matches the context

Reply:
That's cool. Elsewhere I noticed you were removing `make_gzipped_json_response()` and just returning the dict. The more of that we do, the more motivated I'll be to [move the compression to the nginx](https://app.asana.com/0/1200435587978125/1201078871707042/f) where it belongs :)

Reply:
That gzip helper doesn't respect the request headers which really confused Phil when he was trying to use `curl` on these endpoints.